### PR TITLE
Code Action: Allow enabling/disabling and improve ranking of explain

### PR DESCRIPTION
**Pull request by umpox**
## Description

- Group the fix and explain actions
- Allow enabling/disabling the actions all together

Enabled:
image

Disabled:
image

Interface:
image

## Test plan

Tested locally, enabling and disabling through interface and manually in VS Code settings

### CODE REVIEW

File: vscode/src/code-actions/fixup.ts [hunk 1]
```diff
@@ -36,7 +36,6 @@ export class FixupCodeAction implements vscode.CodeActionProvider {
             title: 'Ask Cody to Fix',
         }
         action.diagnostics = diagnostics
-        action.isPreferred = true
```
Comment (umpox) [inline vscode/src/code-actions/fixup.ts:39]: Doesn't affect our ranking, just means users can't trigger this automatically with a keyboard shortcut (wasn't happening anyway AFAICT)

File: vscode/src/code-actions/explain.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ export class ExplainCodeAction implements vscode.CodeActionProvider {
             arguments: [instruction, range],
             title: 'Ask Cody to Explain',
         }
+        action.diagnostics = diagnostics
```
Comment (umpox) [inline vscode/src/code-actions/explain.ts:30]: Means that the code action will appear right alongside the fix action, no other impact

**Review by abeatrix — CHANGES_REQUESTED** [review]
Currently `Ask Cody to Explain` does nothing when inline chat is disabled. Should we show a pop-up or fall back to sidebar chat when inline chat is not enabled?


https://github.com/sourcegraph/cody/assets/68532117/5a902304-c21b-4eee-8be9-bb7065669c87

File: vscode/src/services/StatusBar.ts [hunk 3]
```diff
@@ -100,6 +100,13 @@ export function createStatusBar(): CodyStatusBar {
                     'cody.inlineChat.enabled',
                     c => c.inlineChat
                 ),
+                createFeatureToggle(
+                    'Code Actions',
+                    undefined,
+                    'Enable Cody to fix or explain errors and warnings, directly in your code.',
```
Comment (toolmantim) [inline vscode/src/services/StatusBar.ts:106]:
```suggestion
                    'Enable Cody fix and explain options in the Quick Fix menu',
```

or:

```suggestion
                    'Enable Cody in the Quick Fix menu for fixing and explaining code errors and warnings',
```

**Review by abeatrix — APPROVED** [review]
awesome! 
Not a block: is this a good place to register code action to listen for inline cofig change? https://github.com/sourcegraph/cody/blob/0176b8c7e69b98569f7c39c92791f3ffd3421eb7/vscode/src/services/InlineController.ts#L76-L89

File: vscode/src/services/StatusBar.ts (hunk 3 above)
Comment (umpox) [reply vscode/src/services/StatusBar.ts:106]: Thanks! I like the first suggestion, will update

**Comment by umpox** [comment]
@abeatrix Thought about it but I think it's OK to just keep the basic implementation for now. Doesn't feel like users will often enable/disable inline chat, and the extra overhead of having to register/deregister the action to ensure we don't have duplicate code actions isn't worth it IMO. We can adjust if we get feedback from users that we need this

**Pull request merged**